r/vignettes/developing.Rmd

```diff
@@ -154,15 +184,18 @@ You'll first call `cmake` to configure the build and then `make install`. For th
 
 ```{bash, save=run & !sys_install}
 cmake \
+  -G Ninja \
   -DCMAKE_INSTALL_PREFIX=$ARROW_HOME \
   -DCMAKE_INSTALL_LIBDIR=lib \
+  -DARROW_BUILD_SHARED=OFF \
+  -DARROW_BUILD_STATIC=ON \
   -DARROW_COMPUTE=ON \
   -DARROW_CSV=ON \
   -DARROW_DATASET=ON \
   -DARROW_EXTRA_ERROR_CONTEXT=ON \
   -DARROW_FILESYSTEM=ON \
   -DARROW_INSTALL_NAME_RPATH=OFF \
-  -DARROW_JEMALLOC=ON \
+  -DARROW_JEMALLOC=OFF \
```

Comment:
This library isn't supported on Windows.

Curiosity: why is this part of the std build and not listed as an optional component below? Is it required on Linux or Mac OS?

Reply:
I would need to double check, but I believe this is the default memory allocator on linux (or it was when we wrote this).

Reply:
Yup, the defaults are (on not-macos): jemalloc, mimalloc, system (in that order, if they are available)
On macos: mimalloc, jemalloc, system (in that order, if they are available)

So on windows (if jemalloc isn't supported) the order will be: mimalloc, system (in that order)

Because of this, I think we _do_ want it here at least when building on linux or macos. Does it fail gracefully if it's `ON` on windows or does it error completely?

Reply:
It doesn't currently fail gracefully; it attempts to build jemalloc on Windows and fails. I spent a good part of a day thinking my Arrow build broken until I realized jemalloc just isn't supported on Windows 😢.

I can make a bolded note here that jemalloc should be off for Windows. And then I'd like add something to CMake (or perhaps elsewhere) to not let you try to build jemalloc.

Reply:
That sounds good